.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,54 @@
+name: Pull Request Validation
+
+on:
+  pull_request:
+    branches:
+      - trunk
+    types:
+      - "labeled"
+      - "unlabeled"
+      - "opened"
+      - "reopened"
+      - "synchronize"
+
+jobs:
+  check-labels:
+    name: Check Labels and Milestone
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Check Type Labels
+        if: |
+          !contains( github.event.pull_request.labels.*.name, '[Type] Bug' ) &&
+          (
+            !contains( github.event.pull_request.labels.*.name, '[Type] Feature' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Type] Documentation' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Type] Discussion' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Type] Enhancement' )
+          )
```

Comment:
Why are all but the first check here in parentheses? That shouldn't be necessary I'd think? It looks a bit confusing since it makes one think the first condition is in any way special, but it's similar to the ones below.

Anyway, I'm wondering whether we could simplify this. This code would be easier to maintain if we only checked whether there is any label that starts with `[Type] ...` - otherwise we would need to update the code every time we'd add a new type label. Maybe we can do a regex or some sort of `startsWith` matching here instead?

Reply:
> Why are all but the first check here in parentheses?

This seems to be how the `.yml` syntax works in GitHub Actions. The other option would have been to have all the expressions on the same line (see https://github.community/t/how-to-write-multi-line-condition-in-if/128477/2).

> Maybe we can do a regex or some sort of startsWith matching here instead?

That's the first option I actually explored 😅. Yes, I agree it'll definitely make the code more maintainable. I'll do another tries with https://github.com/marketplace/actions/actions-ecosystem-action-regex-match.

Reply:
Maybe we should use something like this? https://github.com/marketplace/actions/label-checker-for-pull-requests